src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.cs

```diff
@@ -411,7 +410,7 @@ internal sealed partial class CultureData
 
         // Cache of regions we've already looked up
         private static volatile Dictionary<string, CultureData>? s_cachedRegions;
-        private static volatile Dictionary<string, string>? s_regionNames;
+        private static Dictionary<string, string>? s_regionNames;
```

Comment:
Why `s_cachedRegions` is still volatile? Is accessing it needs to be ordered with accessing something else?

Reply:
From the use it looks like just a shared cache. We use a lock when querying/adding to the cache, since it is an ordinary Dictionary. 
We initialize the static without any locks though in a "last to assign wins" fashion. Which makes sense if the cache itself has no identity whatsoever and can be cheaply refilled. It seems doubtful that anyone cares about the order of accesses to `s_cachedRegions`.

Reply:
I'd only focused on the obvious ones. We can incrementally remove more.

Reply:
There is an interesting scenario when we set `s_cachedRegions = null` in `ClearCachedData`, I assume when somehting changed machine-wide.
If that runs concurrently with a thread doing `s_cachedRegions = tempHashTable`, we could see that installing a cache with data from before the `ClearCachedData`.

I am not familiar with what is happening here at the high level. Perhaps this clearing is coordinated in a way that this kind of race cannot happen...   
Either way `volatile` will not help with that.